internal/controller/pipelinerun_helper_test.go

```diff
@@ -0,0 +1,301 @@
+package controller
+
+import (
+	"encoding/json"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	tektonv1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	v1 "knative.dev/pkg/apis/duck/v1"
+)
+
+var _ = Describe("Unit testing for pipelinerun_helper", func() {
+	var (
+		testTruePipelineRun,
+		testFalsePipelineRun,
+		testErrMmarshelPipelineRun *tektonv1.PipelineRun
+		notPipelineRun *tektonv1.TaskRun
+	)
+	const (
+		pushPipelineRunName        = "push-pipelinerun-sample"
+		pullRequestPipelineRunName = "pull-request-pipelinerun-sample"
+		namespace                  = "default"
+	)
+
+	Describe("Push Pipelinerun; Ended; with finalizer, annotation and results", func() {
+		testTruePipelineRun = &tektonv1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pushPipelineRunName,
+				Namespace: namespace,
+				Labels: map[string]string{
+					PipelineRunTypeLabel: PushPipelineRunTypeValue,
+				},
+				Finalizers: []string{
+					NotificationPipelineRunFinalizer,
+				},
+				Annotations: map[string]string{
+					NotificationPipelineRunAnnotation: NotificationPipelineRunAnnotationValue,
+				},
+			},
+			Spec: tektonv1.PipelineRunSpec{
+				PipelineRef: &tektonv1.PipelineRef{},
+			},
+			Status: tektonv1.PipelineRunStatus{
+				PipelineRunStatusFields: tektonv1.PipelineRunStatusFields{
+					StartTime:      &metav1.Time{Time: time.Now()},
+					CompletionTime: &metav1.Time{Time: time.Now().Add(5 * time.Minute)},
+					Results: []tektonv1.PipelineRunResult{
+						{
+							Name:  "IMAGE_DIGEST",
+							Value: *tektonv1.NewStructuredValues("image_digest_value"),
+						},
+						{
+							Name:  "IMAGE_URL",
+							Value: *tektonv1.NewStructuredValues("image"),
+						},
+						{
+							Name:  "CHAINS-GIT_URL",
+							Value: *tektonv1.NewStructuredValues("git_url_value"),
+						},
+						{
+							Name:  "CHAINS-GIT_COMMIT",
+							Value: *tektonv1.NewStructuredValues("git_commit_value"),
+						},
+					},
+				},
+				Status: v1.Status{
+					Conditions: v1.Conditions{
+						apis.Condition{
+							Message: "Tasks Completed: 12 (Failed: 0, Cancelled 0), Skipped: 2",
+							Reason:  "Completed",
+							Status:  "True",
+							Type:    apis.ConditionSucceeded,
+						},
+					},
+				},
+			},
+		}
+		Context("when a push pipelinerun is created, end successfully and has results", func() {
```

Comment:
We might not need the context, as I'm not sure it adds any new information, but if we keep it, I think it should say something like this:
```suggestion
		Context("when a push pipelinerun ends successfully and has results", func() {
```

Reply:
Done